### ISSUE TITLE
A renderMonth function

### ISSUE THREAD

**Issue by bentaylorhk** [mui-org/material-ui-pickers#1170]
Is there any possibility of an implementation of a renderMonth function? This would work very similarly to how the renderDay function works, although allowing custom rendering for different selected months.

**Comment by rdbmax** [comment]
Yes it could be usefull in order to implement quarter selection.

**Comment by oliviertassinari** [comment]
Do you have more details on the end result you wish to accomplish with such feature?

**Comment by phsantiago** [comment]
I think I need the same, from my point of view, we could try a render prop here https://github.com/mui-org/material-ui-pickers/blob/fd86dc2daa19ee8d85240b170000c29c6c5cb54a/lib/src/views/Calendar/Month.tsx#L70
To be possible to add more components in month rendering, like it's possible with renderDay.

**Comment by oliviertassinari** [comment]
Thanks for opening this issue. I agree, It seems that we would benefit from more customization power. We will have the same problem to solve on the more advanced components that can't allow a composition API. Moving the discussion to https://github.com/mui-org/material-ui/issues/21453.

**Comment by oliviertassinari** [comment]
At this point, I think that we could consider having a shared component to display:

- Years
- Quarters
- Months

I fail to envision why we would need a different component for each.
